File: doubly_linked_list/recipe-01/python/test_doubly_linked_list.py
from doubly_linked_list import DoublyLinkedList

def test():
    dll = DoublyLinkedList()
    dll.append("foo")
    dll.append("bar")
    dll.append("biz")
    dll.append("whew")
    print("Items in List : ")
    dll.print_foward()
    print("List after deleting node with data whew")
    dll.delete("whew")
    dll.print_foward()
    
    print("List count: {}".format(dll.count))
    print("Print list backwards")
    dll.print_backward()
    
    print("Reverse list ")
    dll.reverse()
    dll.print_foward()
    
    print("Append item to front of list")
    dll.insert_head(55)
    dll.print_foward()
    
    print("Get First element: {}".format(dll[0]))

if __name__ == "__main__":
    test()

File: doubly_linked_list/recipe-01/c/test_doubly_linked_list.c
#include "string_list.h"

void test()
{
    DoublyLinkedList* dll = NewList();
    ListAppend(dll, "foo");
    ListAppend(dll, "bar");
    ListAppend(dll, "biz");
    ListAppend(dll, "whew");

    printf("Items in List : \n");
    ListPrintFoward(dll);

    printf("List after deleting node with data whew\n");
    ListDelete(dll, "whew");
    ListPrintFoward(dll);

    printf("List count: %d\n", ListCount(dll));
    printf("Print list backwards\n");
    ListPrintBackward(dll);

    printf("Reverse list \n" );
    ListReverse(dll);
    ListPrintFoward(dll);

    printf("Append item to front of list\n");
    ListInsertHead(dll, "55");
    ListPrintFoward(dll);

    printf("Get First element: %s\n", ListGetItem(dll, 0));

    FreeList(dll);
}

int main()
{
    Init();

    test();

    return 0;
}

